packages/truffle-box/lib/utils/unbox.js

```diff
@@ -1,90 +1,73 @@
-const fs = require("fs-extra");
+const fse = require("fs-extra");
 const path = require("path");
 const ghdownload = require("github-download");
-const request = require("request");
+const rp = require("request-promise-native");
 const vcsurl = require("vcsurl");
 const parseURL = require("url").parse;
-const exec = require("child_process").exec;
+const execSync = require("child_process").execSync;
 const inquirer = require("inquirer");
 
-function verifyURL(url) {
+async function verifyURL(url) {
   // Next let's see if the expected repository exists. If it doesn't, ghdownload
   // will fail spectacularly in a way we can't catch, so we have to do it ourselves.
-  return new Promise(function(accept, reject) {
-    var configURL = parseURL(
-      vcsurl(url)
-        .replace("github.com", "raw.githubusercontent.com")
-        .replace(/#.*/, "") + "/master/truffle-box.json"
-    );
-
-    var options = {
-      method: "HEAD",
-      uri: "https://" + configURL.host + configURL.path
-    };
-    request(options, function(error, r) {
-      if (error) {
-        return reject(
-          new Error(
-            "Error making request to " +
-              options.uri +
-              ". Got error: " +
-              error.message +
-              ". Please check the format of the requested resource."
-          )
-        );
-      } else if (r.statusCode === 404) {
-        return reject(
-          new Error(
-            "Truffle Box at URL " +
-              url +
-              " doesn't exist. If you believe this is an error, please contact Truffle support."
-          )
-        );
-      } else if (r.statusCode !== 200) {
-        return reject(
-          new Error(
-            "Error connecting to github.com. Please check your internet connection and try again."
-          )
-        );
-      }
-      accept();
-    });
-  });
+  const configURL = parseURL(
+    `${vcsurl(url)
+      .replace("github.com", "raw.githubusercontent.com")
+      .replace(/#.*/, "")}/master/truffle-box.json`
+  );
+
+  const options = {
+    method: "HEAD",
+    uri: `https://${configURL.host}${configURL.path}`,
+    resolveWithFullResponse: true,
+    simple: false
+  };
+
+  try {
```

Comment:
I would say that it looks like this try/catch can be removed as well as it does not seem to do anything.

Reply:
Under certain conditions the request fails before returning a `statusCode`. I imagine we probably want to keep a catch there _just in case_.